admin/includes/functions/general.php

```diff
@@ -738,23 +738,30 @@ function zen_set_ezpage_status(int $pages_id, int $status, string $status_field)
  */
 function zen_get_orders_status_pulldown_array()
 {
-    global $db;
+    $ordersStatus = zen_getOrdersStatuses();
+    return $ordersStatus['orders_statuses'];
+}
 
+function zen_getOrdersStatuses(bool $keyed = false): array
+{
+    global $db;
+    $orders_statuses = [];
     $orders_status_array = [];
-    $orders_status = $db->Execute("SELECT orders_status_id, orders_status_name
-                                   FROM " . TABLE_ORDERS_STATUS . "
-                                   WHERE language_id = " . (int)$_SESSION['languages_id'] . "
-                                   ORDER BY orders_status_id");
-
-    while (!$orders_status->EOF) {
-        $orders_status_array[] = [
-            'id' => $orders_status->fields['orders_status_id'],
-            'text' => $orders_status->fields['orders_status_name']
-        ];
-        $orders_status->MoveNext();
+    $orders_status_query = $db->Execute('SELECT orders_status_id, orders_status_name FROM ' . TABLE_ORDERS_STATUS . '
+                                 WHERE language_id = "' . (int)$_SESSION['languages_id'] . '" ORDER BY sort_order, orders_status_id');
```

Comment:
Would it make sense to include the caching parameters here, as used in the dashboard widget?

`$db->Execute(SQLQUERY, false, true, 3600)` // for one hour

Reply:
i'm going to wait an hour before i answer this...

Reply:
i am really on the fence on this one.  i do not feel strongly 1 way or another.  i think maintaining order statuses is rarely done.  but i also do not think that caching this query saves much time in the grand scope of things.  i could see people questioning why their statuses are not showing up right after entering them and wanting to change an order status.

that said, i could see adding another optional input parm to effect caching on the method call from the widget.

but i am open.

Reply:
Understood. The only time the cache would need a reset would be directly after an edit.
And the reset has to be done to the call to the exact same query string
So yes passing a parameter, maybe `$resetCache=false`, would make sense ... and that would only ever be passed as true from a fresh call to this same function immediately after an insert/edit/delete and before the usual zen_redirect that follows such actions.

Reply:
I also agree that this is not an expensive query so the db's own internal caching is probably sufficient.

Reply:
Will leave it for now